vavr/src/main/java/org/jdbi/v3/vavr/VavrTupleRowMapperFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.vavr;
+
+import io.vavr.CheckedFunction1;
+import io.vavr.Tuple;
+import io.vavr.Tuple0;
+import io.vavr.Tuple1;
+import io.vavr.Tuple2;
+import io.vavr.Tuple3;
+import io.vavr.Tuple4;
+import io.vavr.Tuple5;
+import io.vavr.Tuple6;
+import io.vavr.Tuple7;
+import io.vavr.Tuple8;
+import io.vavr.collection.Array;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.mapper.ColumnMappers;
+import org.jdbi.v3.core.mapper.NoSuchMapperException;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.mapper.RowMapperFactory;
+import org.jdbi.v3.core.mapper.SingleColumnMapper;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+import static org.jdbi.v3.core.generic.GenericTypes.resolveType;
+
+public class VavrTupleRowMapperFactory implements RowMapperFactory {
+
+    @Override
+    public Optional<RowMapper<?>> build(Type type, ConfigRegistry config) {
+        Class<?> erasedType = getErasedType(type);
+        if (Tuple0.class.equals(erasedType) || Tuple.class.equals(erasedType)) {
+            return Optional.of((rs, ctx) -> Tuple.empty());
+        }
+
+        if (type instanceof ParameterizedType && canBeMappedToTuple(erasedType)) {
+            Class<? extends Tuple> tupleClass = (Class<? extends Tuple>) erasedType;
+            final Array<RowMapper<?>> colMappers = Array.of(tupleClass.getTypeParameters())
+                    .map(tp -> resolveType(tp, type))
+                    .zipWithIndex()
+                    .map(t -> getColumnMapper(t._1, t._2, config));
+
+            return Optional.of((rs, ctx) ->
+                    buildTuple(tupleClass, i -> colMappers.get(i).map(rs, ctx)));
```

Comment:
It does make sense, and after re-reading it again and again it might actually be that the implementation for this is rather trivial - or better - analogue to `MapEntryMapper`. I still would like your feedback on the `TupleMappers` class since it has an interaction with global configuration. Once this is done I will start the implementation on this

Reply:
Did my earlier comment on `TupleMappers` delegating vs duplicating `MapEntryMappers` answer your question?

Reply:
Ough there is a problem here I fear. The proposed sanity check woll not work since i have no way to find out if type T of Tuple<X, T, Y> cannot be rowmapped since every primitive type yields a row mapper on the config.

not sure if I expressed myself correctly. I will push my status and you may have a look at the test `testMapTuple3WithoutSpecifiedColumn_shouldFail()` where the _3 yields the first column in the query because nothing is specified and there IS a rowmapper for this type... just not the one we might like

Reply:
Yeah, I've been wondering if that fail over to column mappers behavior should live in `RowMappers.findFor`. This current design has come up as a problem elsewhere.

Maybe `findFor` should only check the explicit row mappers, and we add a separate `resolveFor` method that fails over to column mappers.

We'd also have to add an equivalent method in `StatementContext`, perhaps `findMapperFor` (i.e. no row or column in the name, therefore it could come from either).